Polish docstrings for static operators

erikkemperman opened the pull request:
I notice that the `generate_with_relative_time` and `generate` static ops don't handle a scheduler parameter, but the docstring says:

https://github.com/ReactiveX/RxPY/blob/dc765aaad01cbb4493c280397ff34333cf2158a3/rx/__init__.py#L491-L494

For the moment I've removed the docstring bit about "using the specified" scheduler. But I wonder if an optional `scheduler` argument should be added here?

The newly added marble diagrams for `case` and `for_in` looked a bit "cramped" to my eyes. So I made them slightly wider, but if I missing some particular reason behind this then please let me know (@MainRo)!

Another question, all references to `Observable` in `rx/__init__.py` are actually to `rx.core.observable.Observable` -- probably just because it is already available as that class is being re-exported `rx.Observable`.

But from an API perspective, should the input Observables not actually be `typing.Observable`, so as not to tie the users to our particular concrete base class?

jcafhe commented on rx/__init__.py:
```diff
@@ -43,11 +47,11 @@ def case(mapper: Callable[[], Any],
+         b----10---20---30---|
         [case(mapper, { 1: a, 2: b })]
-        ---1--2--3--4--|
+        -----1----2----3----4---|
```
I think the resulting observable should be shifted to the right (+ 3 hyphens), i.e. the 'a' observable is subscribed when the source observable emits `1`.

erikkemperman replied:
```
        ---1---|
        a----1----2----3----4---|
         b----10---20---30---|
        [case(mapper, { 1: a, 2: b })]
        --------1----2----3----4---|
```

Do you mean like this? I don't know, I tried to keep the vertical alignment the same... I think the point is to emphasize that the result sequence is the same as `a`. But I got this wrong before :-)

jcafhe commented on rx/__init__.py:
```diff
@@ -110,35 +118,43 @@ def catch_with_iterable(sources: Iterable[Observable]) -> Observable:
         ---1---2---3---4---|
 
+    Args:
+        subscribe: [Optional] Subscription function.
```
Should `subscribe` be qualified as `Optional` with `None` as a default value in the signature? Don't know if it's worth to call `create` with no subscribe function thougth.

erikkemperman replied:
Ah, good catch. Sloppy/paste, I guess... I'll amend.

jcafhe commented on rx/__init__.py:
```diff
@@ -246,47 +275,56 @@ def empty(scheduler: Optional[typing.Scheduler] = None) -> Observable:
+         b----10---20--|
         [for_in((a, b), lambda i: i+1)]
-        ---2--3--11--21-|
+        -----2----3----11----21--|
```
Not sure here, but I think there should be 6 hyphens instead of 4 between `3` and `11`, i.e. 2 from the end of 'a' observable + 4 from the start of the 'b' observable.

erikkemperman replied:
```
        a----1----2--|
         b----10---20--|
        [for_in((a, b), lambda i: i+1)]
        -----2----3------11----21--|
```
Like this?

jcafhe replied:
Exactly :+1:

jcafhe commented on rx/__init__.py:
```diff
@@ -997,9 +1056,8 @@ def using(resource_factory: Callable[[], typing.Disposable],
-    observable sequence produces an element. The observables can be
-    passed either as separate arguments or as a list.
+    sequence by creating a :class:`tuple` only when the first
+    observable sequence produces an element..
```
Trailing period. Sorry to bother you :trollface:

erikkemperman replied:
No bother, thanks!

jcafhe commented:
Good job! :+1:  

> For the moment I've removed the docstring bit about "using the specified" scheduler. But I wonder if an optional scheduler argument should be added here?

Yes, that could be great, just by adding the common 'pattern' 
`_scheduler = scheduler or scheduler_ or default_scheduler`
 

> But from an API perspective, should the input Observables not actually be typing.Observable, so as not to tie the users to our particular concrete base class?

I agree too.

erikkemperman commented:
OK, I've removed the commit about the marble diagrams for now -- I guess I would prefer if @MainRo could take a look, if there's a bit of time, at making them slightly more "balanced" (rather than me risking inadvertently changing something in his effort). It's a really minor point, obviously.

And I think I've addressed the other remarks, about trailing period and incorrect `Optional`.

I'll see if I can quickly propose something about using `typing.Observable` where appropriate, as well as adding `scheduler` arguments where it might make sense.
